File: Java/LeetCode/Count Almost Equal Pairs I.java
// https://leetcode.com/problems/count-almost-equal-pairs-i/

class Solution {

    public int countPairs(int[] nums) {
        int count = 0;

        for (int i = 0; i < nums.length; ++i) {
            for (int j = i+1; j < nums.length; ++j) {
                if (nums[i] == nums[j]) {
                    count++;
                    continue;
                } else {
                    String x = String.valueOf(nums[i] > nums[j] ? nums[i] : nums[j]);
                    String y = String.valueOf(nums[i] > nums[j] ? nums[j] : nums[i]);
                    for (int m = 0; m < x.length(); ++m) {
                        for (int n = m+1; n < x.length(); ++n) {
                            StringBuilder xSwapped = new StringBuilder(x);
                            xSwapped.setCharAt(m, x.charAt(n));
                            xSwapped.setCharAt(n, x.charAt(m));
                            if (Integer.parseInt(xSwapped.toString()) == Integer.parseInt(y)) {
                                count++;
                            }
                        }
                    }
                }
            }
        }

        return count;
    }

}

File: Kotlin/LeetCode/Count Almost Equal Pairs I.kt
// https://leetcode.com/problems/count-almost-equal-pairs-i

class Solution {
    
    fun countPairs(nums: IntArray): Int {
        var count = 0;

        for (i in 0 until nums.size) {
            for (j in i+1 until nums.size) {
                if (nums[i] == nums[j]) {
                    count++;
                    continue;
                }

                val x = if (nums[i] > nums[j]) nums[i].toString() else nums[j].toString();
                val y = if (nums[i] > nums[j]) nums[j] else nums[i];
                
                var matched = false;
                for (m in 0 until x.length) {
                    if (matched) break;
                    for (n in m+1 until x.length) {
                        if (matched) break;
                        val xSwapped = StringBuilder(x);
                        xSwapped[m] = x[n];
                        xSwapped[n] = x[m];
                        if (xSwapped.toString().toInt() == y) {
                            count++;
                            matched = true;
                        }
                    }
                }
            }
        }

        return count;
    }
    
}
